login/transport.go

```diff
@@ -0,0 +1,85 @@
+package login
+
+// transport.go contains the binding from endpoints to a concrete transport.
+// In our case we just use a REST-y HTTP transport.
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+	httptransport "github.com/go-kit/kit/transport/http"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+)
+
+// MakeHTTPHandler mounts the endpoints into a REST-y HTTP handler.
+func MakeHTTPHandler(ctx context.Context, e Endpoints, logger log.Logger) http.Handler {
+	r := mux.NewRouter().StrictSlash(false)
+	options := []httptransport.ServerOption{
+		httptransport.ServerErrorLogger(logger),
+		httptransport.ServerErrorEncoder(encodeError),
+	}
+
+	// GET /login       Login
+	// GET /register    Register
+
+	r.Methods("GET").Path("/login").Handler(httptransport.NewServer(
+		ctx,
+		e.LoginEndpoint,
+		decodeLoginRequest,
+		encodeResponse,
+		options...,
+	))
+	r.Methods("GET").Path("/register").Handler(httptransport.NewServer(
+		ctx,
+		e.RegisterEndpoint,
+		decodeRegisterRequest,
+		encodeResponse,
+		options...,
+	))
+
+	return r
+}
+
+func encodeError(_ context.Context, err error, w http.ResponseWriter) {
+	code := http.StatusInternalServerError
+	switch err {
+	case ErrNotAuthorized:
+		code = http.StatusUnauthorized
+	}
+	w.WriteHeader(code)
+	json.NewEncoder(w).Encode(map[string]interface{}{
```

Comment:
Probably want to set a Content-Type here. (I missed this in cataloguesvc! Updating now.)